tests/test_quantum_gradients.py

```diff
@@ -625,22 +625,6 @@ def circuit(reused_param, other_param):
 
                 assert grad_eval == pytest.approx(grad_true, abs=tol)
 
-    def test_gradient_exception_on_sample(self):
```

Comment:
How come this gets deleted?

Reply:
The new QNode doesn't raise an error when attempting to use parameter-shift on `qml.sample` 😬 

I must admit, I felt really bad keeping that validation since:

- If you remove the error, all the logic works 💯, and you get the single-shot parameter-shift estimates of the gradient.

- It is _so much faster_ than shot batching with `expval` (e.g., doing `shots=([1, 100])`), since that involves a lot of tensor manipulation, but gives the exact same result.

Recently, a lot of users have been asking for this functionality on the forum, and I feel bad blocking it when... it works 😬 

What do you think?